app/helpers/basespace_helper.rb

```diff
@@ -0,0 +1,122 @@
+require 'open-uri'
+
+BASESPACE_CURRENT_PROJECTS_URL = "https://api.basespace.illumina.com/v1pre3/users/current/projects".freeze
+BASESPACE_PROJECT_DATASETS_URL = "https://api.basespace.illumina.com/v2/projects/%s/datasets".freeze
+BASESPACE_DATASET_FILES_URL = "https://api.basespace.illumina.com/v2/datasets/%s/files?filehrefcontentresolution=true".freeze
+# 1024 is the maximum limit allowed by Basespace.
+# If users reach this limit, we will need to implement multiple requests to fetch all the samples.
+BASESPACE_PAGE_SIZE = 1024
+
+module BasespaceHelper
+  def basespace_projects(access_token)
+    response = HttpHelper.get_json(
+      BASESPACE_CURRENT_PROJECTS_URL,
+      { limit: BASESPACE_PAGE_SIZE },
+      "Authorization" => "Bearer #{access_token}"
+    )
+
+    if response.nil? || response["Response"].nil? || response["Response"]["Items"].nil?
```

Comment:
you could also do something like `response&.dig("Response", "Items").nil?`

Reply:
Wow!